content/00.build/30.zksync-sso/10.architecture.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Architecture
+description: Learn about architectural components of ZKsync SSO and how they work together.
+---
+
+The ZKsync SSO consists of three components that work together to offer a seamless experience for both developers and users.
+
+## ZKsync SSO Developer SDK
+
+The **Developer SDK** is a client-side library that developers can integrate into their applications with minimal effort.
+It handles account authentication, session management, and spend limit controls,
+allowing developers to enhance security and streamline the user experience, without dealing with complex wallet and key management infrastructure.
+
+**Features:**
+
+- **Authentication:** Enables users to authenticate using passkeys, supporting biometric methods like Face ID and fingerprint recognition.
+- **Session and Spend Limit Management:** Allows applications to manage user sessions with specific token limits,
+enhancing security by restricting the scope and usage of session keys.
+
+**Platforms:**
+
+- **Web SDK:** Available now, implemented in JavaScript for web applications.
+- **Swift SDK:** For iOS applications (*coming soon*).
+- **Kotlin SDK:** For Android applications (*coming soon*).
+
+### ZKsync SSO Auth Server
+
+The **Auth Server** is a stateless, off-chain hosted service (provided by ZKsync as a public good) that facilitates key aspects of the authentication process.
+It acts as an intermediary between the client application and the ZKsync chain, handling:
+
+- **Passkey Creation:** Assists in generating and managing user passkeys for authentication.
+- **Session Creation:** Manages user sessions, including session keys and their associated permissions.
+- **Transaction Signatures:** Signs transactions on behalf of the user within the constraints of their session limits.
+
+This server ensures that sensitive operations are handled securely and efficiently without exposing
+private keys or requiring users to manage complex cryptographic details.
+
+## Smart Contracts
+
+A set of smart contracts deployed on the ZKsync chain manage accounts, authentication modules, and session keys.
+These contracts form the on-chain backbone of the ZKsync SSO system.
+
+- **ERC7579Account.** Implements the ERC-7579 standard for modular smart contract accounts, allowing for extensible account functionalities.
+- **AAFactory.** A factory contract used to deploy new user accounts on the chain efficiently.
+- **SessionLimitSessionKeyModule.** Manages session keys with specific spend limits, enhancing security by restricting their scope and usage.
```

Comment:
Is this still the name of the module, I thought @ly0va updated it?

Reply:
Good point, changing this to SessionKey - @ly0va please flag if incorrect

Reply:
It's `SessionKeyValidator` now

Reply:
Updated